SingularityService/src/main/java/com/hubspot/singularity/data/DisasterManager.java

```diff
@@ -0,0 +1,146 @@
+  }
+
+  private String getActionPath(SingularityDisabledActionType action) {
+    return ZKPaths.makePath(DISABLED_ACTIONS_PATH, action.name());
+  }
+
+  public boolean isDisabled(SingularityDisabledActionType action) {
+    return exists(getActionPath(action));
+  }
+
+  public SingularityDisabledAction getDisabledAction(SingularityDisabledActionType action) {
+    Optional<SingularityDisabledAction> maybeDisabledAction = getData(getActionPath(action), disabledActionTranscoder);
+    return maybeDisabledAction.or(new SingularityDisabledAction(action, String.format(MESSAGE_FORMAT, action, DEFAULT_MESSAGE), Optional.<String>absent(), false));
+  }
+
+  public SingularityCreateResult disable(SingularityDisabledActionType action, Optional<String> maybeMessage, Optional<SingularityUser> user, boolean systemGenerated) {
+    SingularityDisabledAction disabledAction = new SingularityDisabledAction(
+      action,
+      String.format(MESSAGE_FORMAT, action, maybeMessage.or(DEFAULT_MESSAGE)),
+      user.isPresent() ? Optional.of(user.get().getId()) : Optional.<String>absent(),
+      systemGenerated);
+
+    return save(getActionPath(action), disabledAction, disabledActionTranscoder);
+  }
+
+  public SingularityDeleteResult enable(SingularityDisabledActionType action) {
+    return delete(getActionPath(action));
+  }
+
+  public List<SingularityDisabledAction> getDisabledActions() {
+    List<String> paths = new ArrayList<>();
+    for (String path : getChildren(DISABLED_ACTIONS_PATH)) {
+      paths.add(ZKPaths.makePath(DISABLED_ACTIONS_PATH, path));
+    }
+
+    return getAsync(DISABLED_ACTIONS_PATH, paths, disabledActionTranscoder);
+  }
+
+  public void addDisaster(SingularityDisasterType disaster) {
+    create(ZKPaths.makePath(ACTIVE_DISASTERS_PATH, disaster.name()));
+  }
+
+  public void removeDisaster(SingularityDisasterType disaster) {
+    delete(ZKPaths.makePath(ACTIVE_DISASTERS_PATH, disaster.name()));
+  }
+
+  public List<SingularityDisasterType> getActiveDisasters() {
+    List<String> disasterNames = getChildren(ACTIVE_DISASTERS_PATH);
+    List<SingularityDisasterType> disasters = new ArrayList<>();
+    for (String name : disasterNames) {
+      disasters.add(SingularityDisasterType.valueOf(name));
+    }
+    return disasters;
+  }
+
+  public void saveDisasterStats(SingularityDisasterStats stats) {
+    save(DISASTER_STATS_PATH, stats, disasterStatsTranscoder);
+  }
+
+  public Optional<SingularityDisasterStats> getDisasterStats() {
+    return getData(DISASTER_STATS_PATH, disasterStatsTranscoder);
+  }
+
+  public void updateActiveDisasters(List<SingularityDisasterType> previouslyActiveDisasters, List<SingularityDisasterType> newActiveDisasters) {
+    for (SingularityDisasterType disaster : previouslyActiveDisasters) {
+      if (!newActiveDisasters.contains(disaster)) {
+        removeDisaster(disaster);
+      }
+    }
+
+    for (SingularityDisasterType disaster : newActiveDisasters) {
+      addDisaster(disaster);
```

Comment:
Ah, yeah I can make it skip those. Originally I was going to store a message in there, so it made sense to call that again in case the message had been updated. Can probably do a `if (!exists(path))` check first, thanks!